### ISSUE TITLE
issue with classpath in Riemann

### ISSUE THREAD

**Issue by konstan** [slipstream/slipstream-auto-scale-apps#2]
At the moment the application doesn't work.

There is an issue with classpath in Riemann - it fails on the libs SS Client API provides as first jar in the classpath. (This is something that we do here https://github.com/slipstream/slipstream-auto-scale-apps/blob/master/orchestrator/deployment.sh#L105).

Putting SS Client API jar as the second one (default in Riemann) - fails in kvlt's dependencies.

TODO: provide the actual stack traces for both cases.

**Comment by konstan** [comment]
When `EXTRA_CLASSPATH` is defined in `/etc/default/riemann` as follows

```
~# cat /etc/default/riemann
EXTRA_CLASSPATH=/opt/slipstream/client/lib/clj-ss-client.jar
RIEMANN_CONFIG=/etc/riemann/riemann-ss-streams.clj
~#
```

and `clj-ss-client.jar` gets downloaded from http://nuv.la/downloads/clj-ss-client.jar.

Then, this is the stacktrace with `JAR="/usr/share/riemann/riemann.jar:$EXTRA_CLASSPATH"` defined in `/usr/bin/riemann`.

```
ERROR [2016-08-29 08:46:05,828] main - riemann.bin - Couldn't start
clojure.lang.Compiler$CompilerException: java.lang.IllegalAccessError: compile-if does not exist, compiling:(taoensso/timbre.clj:1:1)
        at clojure.lang.Compiler.load(Compiler.java:7391)
        at clojure.lang.RT.loadResourceScript(RT.java:372)
        at clojure.lang.RT.loadResourceScript(RT.java:363)
        at clojure.lang.RT.load(RT.java:453)
        at clojure.lang.RT.load(RT.java:419)
        at clojure.core$load$fn__5677.invoke(core.clj:5893)
        at clojure.core$load.invokeStatic(core.clj:5892)
        at clojure.core$load.doInvoke(core.clj:5876)
        at clojure.lang.RestFn.invoke(RestFn.java:408)
        at clojure.core$load_one.invokeStatic(core.clj:5697)
        at clojure.core$load_one.invoke(core.clj:5692)
        at clojure.core$load_lib$fn__5626.invoke(core.clj:5737)
        at clojure.core$load_lib.invokeStatic(core.clj:5736)
        at clojure.core$load_lib.doInvoke(core.clj:5717)
        at clojure.lang.RestFn.applyTo(RestFn.java:142)
        at clojure.core$apply.invokeStatic(core.clj:648)
        at clojure.core$load_libs.invokeStatic(core.clj:5774)
        at clojure.core$load_libs.doInvoke(core.clj:5758)
        at clojure.lang.RestFn.applyTo(RestFn.java:137)
        at clojure.core$apply.invokeStatic(core.clj:648)
        at clojure.core$require.invokeStatic(core.clj:5796)
        at clojure.core$require.doInvoke(core.clj:5796)
        at clojure.lang.RestFn.invoke(RestFn.java:482)
        at kvlt.middleware.util$eval6789$loading__5569__auto____6790.invoke(util.cljc:1)
        at kvlt.middleware.util$eval6789.invokeStatic(util.cljc:1)
        at kvlt.middleware.util$eval6789.invoke(util.cljc:1)
        at clojure.lang.Compiler.eval(Compiler.java:6927)
        at clojure.lang.Compiler.eval(Compiler.java:6916)
        at clojure.lang.Compiler.load(Compiler.java:7379)
        at clojure.lang.RT.loadResourceScript(RT.java:372)
        at clojure.lang.RT.loadResourceScript(RT.java:363)
        at clojure.lang.RT.load(RT.java:453)
        at clojure.lang.RT.load(RT.java:419)
        at clojure.core$load$fn__5677.invoke(core.clj:5893)
        at clojure.core$load.invokeStatic(core.clj:5892)
        at clojure.core$load.doInvoke(core.clj:5876)
        at clojure.lang.RestFn.invoke(RestFn.java:408)
        at clojure.core$load_one.invokeStatic(core.clj:5697)
        at clojure.core$load_one.invoke(core.clj:5692)
        at clojure.core$load_lib$fn__5626.invoke(core.clj:5737)
        at clojure.core$load_lib.invokeStatic(core.clj:5736)
        at clojure.core$load_lib.doInvoke(core.clj:5717)
        at clojure.lang.RestFn.applyTo(RestFn.java:142)
        at clojure.core$apply.invokeStatic(core.clj:648)
        at clojure.core$load_libs.invokeStatic(core.clj:5774)
        at clojure.core$load_libs.doInvoke(core.clj:5758)
        at clojure.lang.RestFn.applyTo(RestFn.java:137)
        at clojure.core$apply.invokeStatic(core.clj:648)
        at clojure.core$require.invokeStatic(core.clj:5796)
        at clojure.core$require.doInvoke(core.clj:5796)
        at clojure.lang.RestFn.invoke(RestFn.java:703)
        at kvlt.middleware$eval6781$loading__5569__auto____6782.invoke(middleware.cljc:1)
        at kvlt.middleware$eval6781.invokeStatic(middleware.cljc:1)
        at kvlt.middleware$eval6781.invoke(middleware.cljc:1)
        at clojure.lang.Compiler.eval(Compiler.java:6927)
        at clojure.lang.Compiler.eval(Compiler.java:6916)
        at clojure.lang.Compiler.load(Compiler.java:7379)
        at clojure.lang.RT.loadResourceScript(RT.java:372)
        at clojure.lang.RT.loadResourceScript(RT.java:363)
        at clojure.lang.RT.load(RT.java:453)
        at clojure.lang.RT.load(RT.java:419)
        at clojure.core$load$fn__5677.invoke(core.clj:5893)
        at clojure.core$load.invokeStatic(core.clj:5892)
        at clojure.core$load.doInvoke(core.clj:5876)
        at clojure.lang.RestFn.invoke(RestFn.java:408)
        at clojure.core$load_one.invokeStatic(core.clj:5697)
        at clojure.core$load_one.invoke(core.clj:5692)
        at clojure.core$load_lib$fn__5626.invoke(core.clj:5737)
        at clojure.core$load_lib.invokeStatic(core.clj:5736)
        at clojure.core$load_lib.doInvoke(core.clj:5717)
        at clojure.lang.RestFn.applyTo(RestFn.java:142)
        at clojure.core$apply.invokeStatic(core.clj:648)
        at clojure.core$load_libs.invokeStatic(core.clj:5774)
        at clojure.core$load_libs.doInvoke(core.clj:5758)
        at clojure.lang.RestFn.applyTo(RestFn.java:137)
        at clojure.core$apply.invokeStatic(core.clj:648)
        at clojure.core$require.invokeStatic(core.clj:5796)
        at clojure.core$require.doInvoke(core.clj:5796)
        at clojure.lang.RestFn.invoke(RestFn.java:805)
        at kvlt.platform.http$eval6775$loading__5569__auto____6776.invoke(http.clj:1)
        at kvlt.platform.http$eval6775.invokeStatic(http.clj:1)
        at kvlt.platform.http$eval6775.invoke(http.clj:1)
        at clojure.lang.Compiler.eval(Compiler.java:6927)
        at clojure.lang.Compiler.eval(Compiler.java:6916)
        at clojure.lang.Compiler.load(Compiler.java:7379)
        at clojure.lang.RT.loadResourceScript(RT.java:372)
        at clojure.lang.RT.loadResourceScript(RT.java:363)
        at clojure.lang.RT.load(RT.java:453)
        at clojure.lang.RT.load(RT.java:419)
        at clojure.core$load$fn__5677.invoke(core.clj:5893)
        at clojure.core$load.invokeStatic(core.clj:5892)
        at clojure.core$load.doInvoke(core.clj:5876)
        at clojure.lang.RestFn.invoke(RestFn.java:408)
        at clojure.core$load_one.invokeStatic(core.clj:5697)
        at clojure.core$load_one.invoke(core.clj:5692)
        at clojure.core$load_lib$fn__5626.invoke(core.clj:5737)
        at clojure.core$load_lib.invokeStatic(core.clj:5736)
        at clojure.core$load_lib.doInvoke(core.clj:5717)
        at clojure.lang.RestFn.applyTo(RestFn.java:142)
        at clojure.core$apply.invokeStatic(core.clj:648)
        at clojure.core$load_libs.invokeStatic(core.clj:5774)
        at clojure.core$load_libs.doInvoke(core.clj:5758)
        at clojure.lang.RestFn.applyTo(RestFn.java:137)
        at clojure.core$apply.invokeStatic(core.clj:648)
        at clojure.core$require.invokeStatic(core.clj:5796)
        at clojure.core$require.doInvoke(core.clj:5796)
        at clojure.lang.RestFn.invoke(RestFn.java:408)
        at riemann.config$eval6283.invokeStatic(riemann-ss-streams.clj:27)
        at riemann.config$eval6283.invoke(riemann-ss-streams.clj:27)
        at clojure.lang.Compiler.eval(Compiler.java:6927)
        at clojure.lang.Compiler.load(Compiler.java:7379)
        at clojure.lang.Compiler.loadFile(Compiler.java:7317)
        at clojure.lang.RT$3.invoke(RT.java:320)
        at riemann.config$include.invokeStatic(config.clj:406)
        at riemann.config$include.invoke(config.clj:384)
        at riemann.bin$_main.invokeStatic(bin.clj:99)
        at riemann.bin$_main.invoke(bin.clj:85)
        at clojure.lang.AFn.applyToHelper(AFn.java:156)
        at clojure.lang.AFn.applyTo(AFn.java:144)
        at riemann.bin.main(Unknown Source)
Caused by: java.lang.IllegalAccessError: compile-if does not exist
        at clojure.core$refer.invokeStatic(core.clj:4119)
        at clojure.core$refer.doInvoke(core.clj:4087)
        at clojure.lang.RestFn.applyTo(RestFn.java:139)
        at clojure.core$apply.invokeStatic(core.clj:648)
        at clojure.core$load_lib.invokeStatic(core.clj:5755)
        at clojure.core$load_lib.doInvoke(core.clj:5717)
        at clojure.lang.RestFn.applyTo(RestFn.java:142)
        at clojure.core$apply.invokeStatic(core.clj:648)
        at clojure.core$load_libs.invokeStatic(core.clj:5774)
        at clojure.core$load_libs.doInvoke(core.clj:5758)
        at clojure.lang.RestFn.applyTo(RestFn.java:137)
        at clojure.core$apply.invokeStatic(core.clj:648)
        at clojure.core$require.invokeStatic(core.clj:5796)
        at clojure.core$require.doInvoke(core.clj:5796)
        at clojure.lang.RestFn.invoke(RestFn.java:457)
        at taoensso.timbre$eval11784$loading__5569__auto____11785.invoke(timbre.clj:1)
        at taoensso.timbre$eval11784.invokeStatic(timbre.clj:1)
        at taoensso.timbre$eval11784.invoke(timbre.clj:1)
        at clojure.lang.Compiler.eval(Compiler.java:6927)
        at clojure.lang.Compiler.eval(Compiler.java:6916)
        at clojure.lang.Compiler.load(Compiler.java:7379)
        ... 259 common frames omitted
```

**Comment by konstan** [comment]
Now with `JAR="$EXTRA_CLASSPATH:/usr/share/riemann/riemann.jar"` Riemann starts as

```
22341 ?        Sl     0:21 java -cp /opt/slipstream/client/lib/clj-ss-client.jar:/usr/share/riemann/riemann.jar riemann.bin start /etc/riemann/riemann-ss-streams.clj
```

logs its startup process

```
root@orchestrator-exoscale-ch-gvacae-3b6-51d4-434b-be4e-c77341e648a5:~# cat /var/log/riemann/riemann.log
INFO [2016-08-29 08:57:57,878] clojure-agent-send-off-pool-0 - riemann.graphite - Connecting to  {:host 127.0.0.1, :port 2003}
INFO [2016-08-29 08:57:57,878] clojure-agent-send-off-pool-1 - riemann.graphite - Connecting to  {:host 127.0.0.1, :port 2003}
INFO [2016-08-29 08:57:57,882] clojure-agent-send-off-pool-2 - riemann.graphite - Connecting to  {:host 127.0.0.1, :port 2003}
INFO [2016-08-29 08:57:57,879] clojure-agent-send-off-pool-3 - riemann.graphite - Connecting to  {:host 127.0.0.1, :port 2003}
INFO [2016-08-29 08:57:57,889] clojure-agent-send-off-pool-3 - riemann.graphite - Connected to 127.0.0.1
INFO [2016-08-29 08:57:57,890] clojure-agent-send-off-pool-0 - riemann.graphite - Connected to 127.0.0.1
INFO [2016-08-29 08:57:57,890] clojure-agent-send-off-pool-2 - riemann.graphite - Connected to 127.0.0.1
INFO [2016-08-29 08:57:57,892] clojure-agent-send-off-pool-1 - riemann.graphite - Connected to 127.0.0.1
INFO [2016-08-29 08:58:21,334] main - riemann.config - Starting 1 scale request processor(s).
WARN [2016-08-29 08:58:21,339] main - riemann.config - Starting 1 scale request processor(s).
ERROR [2016-08-29 08:58:21,339] main - riemann.config - Starting 1 scale request processor(s).
INFO [2016-08-29 08:58:21,627] clojure-agent-send-off-pool-0 - riemann.transport.websockets - Websockets server 0.0.0.0 5556 online
INFO [2016-08-29 08:58:21,896] clojure-agent-send-off-pool-2 - riemann.transport.udp - UDP server 0.0.0.0 5555 16384 -1 online
INFO [2016-08-29 08:58:22,042] clojure-agent-send-off-pool-3 - riemann.transport.tcp - TCP server 0.0.0.0 5555 online
INFO [2016-08-29 08:58:22,045] main - riemann.core - Hyperspace core online
root@orchestrator-exoscale-ch-gvacae-3b6-51d4-434b-be4e-c77341e648a5:~#
```

but then crashes producing no stacktarce.

Looking into this now.

**Comment by konstan** [comment]
Had to reboot the VM for riemann to resume producing logs.  Now with SS Client API as the first in the classpath

```
java -cp /opt/slipstream/client/lib/clj-ss-client.jar:/usr/share/riemann/riemann.jar riemann.bin start /etc/riemann/riemann-ss-streams.clj
```

riemann fails with the following stacktrace.

```
INFO [2016-08-29 09:12:16,267] main - riemann.core - Hyperspace core online
WARN [2016-08-29 09:12:16,555] aleph-netty-client-event-pool-4-2 - aleph.http.client - error in HTTP client
io.netty.handler.codec.DecoderException: java.lang.RuntimeException: Unexpected error: java.security.InvalidAlgorithmParameterException: the trustAnchors parameter must be non-empty
        at io.netty.handler.codec.ByteToMessageDecoder.callDecode(ByteToMessageDecoder.java:418)
        at io.netty.handler.codec.ByteToMessageDecoder.channelRead(ByteToMessageDecoder.java:245)
        at io.netty.channel.ChannelHandlerInvokerUtil.invokeChannelReadNow(ChannelHandlerInvokerUtil.java:83)
        at io.netty.channel.DefaultChannelHandlerInvoker.invokeChannelRead(DefaultChannelHandlerInvoker.java:154)
        at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:354)
        at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:145)
        at io.netty.channel.DefaultChannelPipeline.fireChannelRead(DefaultChannelPipeline.java:1078)
        at io.netty.channel.nio.AbstractNioByteChannel$NioByteUnsafe.read(AbstractNioByteChannel.java:125)
        at io.netty.channel.nio.NioEventLoop.processSelectedKey(NioEventLoop.java:510)
        at io.netty.channel.nio.NioEventLoop.processSelectedKeysOptimized(NioEventLoop.java:467)
        at io.netty.channel.nio.NioEventLoop.processSelectedKeys(NioEventLoop.java:381)
        at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:353)
        at io.netty.util.concurrent.SingleThreadEventExecutor$5.run(SingleThreadEventExecutor.java:742)
        at io.netty.util.concurrent.DefaultThreadFactory$DefaultRunnableDecorator.run(DefaultThreadFactory.java:137)
        at java.lang.Thread.run(Thread.java:745)
Caused by: java.lang.RuntimeException: Unexpected error: java.security.InvalidAlgorithmParameterException: the trustAnchors parameter must be non-empty
        at sun.security.ssl.Handshaker.checkThrown(Handshaker.java:1429)
        at sun.security.ssl.SSLEngineImpl.checkTaskThrown(SSLEngineImpl.java:535)
        at sun.security.ssl.SSLEngineImpl.readNetRecord(SSLEngineImpl.java:813)
        at sun.security.ssl.SSLEngineImpl.unwrap(SSLEngineImpl.java:781)
        at javax.net.ssl.SSLEngine.unwrap(SSLEngine.java:624)
        at io.netty.handler.ssl.SslHandler.unwrap(SslHandler.java:1080)
        at io.netty.handler.ssl.SslHandler.unwrap(SslHandler.java:958)
        at io.netty.handler.ssl.SslHandler.decode(SslHandler.java:895)
        at io.netty.handler.codec.ByteToMessageDecoder.callDecode(ByteToMessageDecoder.java:387)
        ... 14 common frames omitted
Caused by: java.lang.RuntimeException: Unexpected error: java.security.InvalidAlgorithmParameterException: the trustAnchors parameter must be non-empty
        at sun.security.validator.PKIXValidator.<init>(PKIXValidator.java:90)
        at sun.security.validator.Validator.getInstance(Validator.java:179)
        at sun.security.ssl.X509TrustManagerImpl.getValidator(X509TrustManagerImpl.java:312)
        at sun.security.ssl.X509TrustManagerImpl.checkTrustedInit(X509TrustManagerImpl.java:171)
        at sun.security.ssl.X509TrustManagerImpl.checkTrusted(X509TrustManagerImpl.java:239)
        at sun.security.ssl.X509TrustManagerImpl.checkServerTrusted(X509TrustManagerImpl.java:136)
        at sun.security.ssl.ClientHandshaker.serverCertificate(ClientHandshaker.java:1496)
        at sun.security.ssl.ClientHandshaker.processMessage(ClientHandshaker.java:216)
        at sun.security.ssl.Handshaker.processLoop(Handshaker.java:979)
        at sun.security.ssl.Handshaker$1.run(Handshaker.java:919)
        at sun.security.ssl.Handshaker$1.run(Handshaker.java:916)
        at java.security.AccessController.doPrivileged(Native Method)
        at sun.security.ssl.Handshaker$DelegatedTask.run(Handshaker.java:1369)
        at io.netty.handler.ssl.SslHandler.runDelegatedTasks(SslHandler.java:1106)
        at io.netty.handler.ssl.SslHandler.unwrap(SslHandler.java:997)
        ... 16 common frames omitted
Caused by: java.security.InvalidAlgorithmParameterException: the trustAnchors parameter must be non-empty
        at java.security.cert.PKIXParameters.setTrustAnchors(PKIXParameters.java:200)
        at java.security.cert.PKIXParameters.<init>(PKIXParameters.java:120)
        at java.security.cert.PKIXBuilderParameters.<init>(PKIXBuilderParameters.java:104)
        at sun.security.validator.PKIXValidator.<init>(PKIXValidator.java:88)
        ... 30 common frames omitted
WARN [2016-08-29 09:12:16,557] aleph-netty-client-event-pool-4-1 - aleph.http.client - error in HTTP client
```

This might be due to version mismatch of `aleph` between what `kvlt` provides and what `riemann` was compiled against and expects to be provided in its jar.

**Comment by konstan** [comment]
Fixed with 057329c1
